pymtl3/passes/rtlir/util/utility.py

```diff
@@ -36,3 +36,19 @@ def get_string( obj ):
     assert arg_name != ''
     comp_name += '__' + arg_name + '_' + get_string(arg_value)
   return comp_name
+
+def get_ordered_upblks( m ):
+  """Return a list of non-update-ff update blocks that have deterministic order"""
+
+  upblks = m.get_update_blocks() - m.get_update_ff()
+  return [ x for x in m.get_update_block_order() if x in upblks ]
+
+def get_ordered_update_ff( m ):
+  """Return a list of update-ff update blocks that have deterministic order"""
+
+  return [ x for x in m.get_update_block_order() if x in m.get_update_ff() ]
+
+def get_ordered_all_upblks( m ):
```

Comment:
is this on purpose not to interleave update_ffs and updates?

Reply:
I don’t think any file in this PR used this feature. But I think it might be useful to keep them apart.

Reply:
hmm can we remove it then?